src/cdomain/value/cdomains/int/defExcDomain.ml

```diff
@@ -469,9 +493,44 @@ struct
         (* If only one of them is bottom, we raise an exception that eval_rv will catch *)
         raise (ArithmeticOnIntegerBot (Printf.sprintf "%s op %s" (show x) (show y))))
 
+  let logor ik x y = norm ik (match x,y with
+    | `Excluded (_, r), `Definite i
+    | `Definite i, `Excluded (_, r) ->
+      if Z.compare i Z.zero >= 0 then
+        `Excluded (S.empty (), R.join r (R.of_interval range_ikind (Int64.zero, Int64.of_int @@ Z.numbits i)))
+      else
+        (match R.minimal r, R.maximal r with
+        | None, _ | _, None -> top_of ik
+        | Some r1, Some r2 -> let b = Int.max (Z.numbits i) (Int64.to_int(Int64.max (Int64.abs r1) (Int64.abs r2))) in
+          `Excluded (S.empty (), R.of_interval range_ikind (Int64.of_int @@ -b, Int64.of_int b))
+        )
+    | `Excluded (_, r1), `Excluded (_, r2) -> `Excluded (S.empty (), R.join r1 r2)
+    | `Definite x, `Definite y ->
+      (try `Definite (Z.logor x y) with | Division_by_zero -> top_of ik)
+    | `Bot, `Bot -> `Bot
+    | _ ->
+      (* If only one of them is bottom, we raise an exception that eval_rv will catch *)
+      raise (ArithmeticOnIntegerBot (Printf.sprintf "%s op %s" (show x) (show y))))
 
-  let logor  = lift2 Z.logor
-  let logxor = lift2 Z.logxor
+  let logxor ik x y = norm ik (match x,y with
+    | `Definite i, `Excluded (_, r)
+    | `Excluded (_, r), `Definite i -> begin
+      if Z.compare i Z.zero >= 0 then
+        `Excluded (S.empty (), R.join r (R.of_interval range_ikind (Int64.zero, Int64.of_int @@ Z.numbits i)))
+      else
+        match R.minimal r, R.maximal r with
+        | None, _ | _, None -> top_of ik
+        | Some r1, Some r2 -> let b = Int.max (Z.numbits i) (Int64.to_int(Int64.max (Int64.abs r1) (Int64.abs r2))) in
+          `Excluded (S.empty (), R.of_interval range_ikind (Int64.of_int @@ -b, Int64.of_int b))
+      end
+    | `Excluded (_, r1), `Excluded (_, r2) -> `Excluded (S.empty (), R.join r1 r2)
+    (* The good case: *)
+    | `Definite x, `Definite y ->
+      (try `Definite (Z.logxor x y) with | Division_by_zero -> top_of ik)
+    | `Bot, `Bot -> `Bot
+    | _ ->
+      (* If only one of them is bottom, we raise an exception that eval_rv will catch *)
+      raise (ArithmeticOnIntegerBot (Printf.sprintf "%s op %s" (show x) (show y))))
```

Comment:
This looks a little like copy pasta :spaghetti:

Reply:
They were at the time, but now there seem to be improvements to various cases of `logxor` which makes them quite different.